include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -223,30 +226,43 @@ pika::future<T> calcTolerance(SizeType i_begin, SizeType i_end, Matrix<const T,
 }
 
 struct TileCollector {
-  SizeType i_begin;
-  SizeType i_end;
+  SizeType idx_begin;
+  SizeType idx_last;
 
 private:
-  template <class T, Device D>
-  std::pair<GlobalTileIndex, GlobalTileSize> getRange(Matrix<const T, D>& mat) {
-    SizeType ntiles = i_end - i_begin + 1;
-    bool is_col_matrix = mat.distribution().size().cols() == 1;
-    SizeType col_begin = (is_col_matrix) ? 0 : i_begin;
+  std::pair<GlobalTileIndex, GlobalTileSize> getRange(const matrix::Distribution& dist) const {
+    SizeType ntiles = idx_last - idx_begin + 1;
+    bool is_col_matrix = dist.size().cols() == 1;
+    SizeType col_begin = (is_col_matrix) ? 0 : idx_begin;
     SizeType col_sz = (is_col_matrix) ? 1 : ntiles;
-    return std::make_pair(GlobalTileIndex(i_begin, col_begin), GlobalTileSize(ntiles, col_sz));
+
+    return std::make_pair(GlobalTileIndex{idx_begin, col_begin}, GlobalTileSize{ntiles, col_sz});
   }
 
 public:
+  auto iteratorLocal(const matrix::Distribution& dist) const {
```

Comment:
Perhaps we can consider a real type for the `tuple<tuple<GlobalTileIndex, GlobalTileIndex>, tuple<GlobalTileIndex, GlobalTileInde>>` here. However, this is absolutely not a blocker for this PR.

Reply:
I'm not super proud of this solution. IMO it is somehow overlapping with the existing `common::iterate_range2d`.

Reply:
`TileCollector` is something we might want to look at during API refactoring, but I would not touch it more than what I need for this PR.

Just want to point out that, given two indices describing a bound included range, it actually encompasses two type of behaviours:
- if it is applied to a column matrix, the given range is considered linear, i.e. over rows of the single column
- otherwise, the given range is considered to be a 2D square

I don't like this very much, but I opted for keeping this functionality and just document it a bit in the code to make it more "transparent".